raddoseLib.py

```diff
@@ -19,8 +20,22 @@ def replaceLine(file,searchExp,replaceExp):
             line = replaceExp
         sys.stdout.write(line)
 
+def replaceLines(filename, replacementStrings):
+    lines = []
+    with open(filename, 'r') as f:
+        for line in f.readlines():
+            for text_to_search, replacement_text in replacementStrings.items():
+                if text_to_search in line:
+                    line = replacement_text
+                    break
+            lines.append(line)
+    
+    with open(filename, 'w') as f:
+        f.writelines(lines)
+
+    
 def run_rd3d(inputFileName):
-    prc = subprocess.Popen(["java", "-jar", os.environ['CONFIGDIR'] + "/raddose3d.jar", "-i", inputFileName, "-p", "rd3d/rd3d_"],
+    prc = subprocess.Popen(["java", "-jar", os.environ['CONFIGDIR'] + "/raddose3d.jar", "-i", inputFileName, "-p", f"rd3d/rd3d_{os.getpid()}_"],
```

Comment:
would it make sense to make the raddose3d.jar file an RPM and deploy? It doesn't seem to belong in the CONFIGDIR or LSDC.

Reply:
I would say so, currently whenever LSDC starts up, it checks if the working dir has a raddose folder, if it doesn't it copies files from the bnlpx_config to the working dir and runs raddose using those files. Probably a historical way of making it run and still being used

Reply:
perhaps we should re-think a lot of what is being done here, but I have proposed a possible solution or two for the raddose RPM issue - https://jira.nsls2.bnl.gov/browse/SB-447